### PR TITLE
Use latest stable and fix master references

### PR DESCRIPTION
**Pull request by pothos**
- Merge the latest stable manifest
- Keep master as default
-  master: Use flatcar-master for portage-stable, reset ebuild repos to master
    The determining combination are the references for scripts, coreos-overlay,
    and portage-stable. All other repositories get referenced via the ebuilds'
    commit IDs.
    Fix the missing reference for portage-stable.
    Remove unnecessary/confusing references to flatcar-master where not relevant.
- Remove build-VERSION.xml which is a duplicate of release.xml    
    To specify that a SDK for the last release should be created,
    people would use `cork --manifest-name release` to overwrite the
    default `default` which links to `master`.
    The copy named `build-2345` is not needed and creates noise if we
    update the manifest for a new release.

# How to use/test
```
$ mkdir sdk-stable
$ cd sdk-stable
$ …/mantle/bin/cork create --manifest-branch kai/latest-stable-and-fix (or if you want to use the latest stable release, add "--manifest-name build-2345")
$ …/mantle/bin/cork enter
$ ./setup_board
$ ./build_packages
$ ./build_image
```

### CODE REVIEW

File: build-2345.xml [hunk 1]
```diff
@@ -6,33 +6,33 @@
...
+  <project groups="minilayout" name="appc/acbuild" path="src/third_party/appc-acbuild" revision="refs/heads/master" upstream="refs/heads/master"/>
+  <project groups="minilayout" name="appc/spec" path="src/third_party/appc-spec" revision="refs/heads/master" upstream="refs/heads/master"/>
+  <project groups="minilayout" name="flatcar-linux/afterburn" path="src/third_party/afterburn" revision="refs/heads/master" upstream="refs/heads/master"/>
+  <project groups="minilayout" name="flatcar-linux/baselayout" path="src/third_party/baselayout" revision="refs/heads/master" upstream="refs/heads/master"/>
+  <project groups="minilayout" name="flatcar-linux/bootengine" path="src/third_party/bootengine" revision="refs/heads/master" upstream="refs/heads/master"/>
+  <project groups="minilayout" name="flatcar-linux/coreos-cloudinit" path="src/third_party/coreos-cloudinit" revision="refs/heads/master" upstream="refs/heads/master"/>
+  <project groups="minilayout" name="flatcar-linux/coreos-overlay" path="src/third_party/coreos-overlay" revision="refs/heads/flatcar-build-2345.3.1" upstream="refs/heads/flatcar-build-2345.3.1"/>
```
Comment (dongsupark) [inline build-2345.xml]: If we set it to `refs/heads/flatcar-build-2345.3.1`, then it means we need to update the branch name every time when we do a new stable release. I am not sure if that's a good idea.
I would rather leave it as `refs/heads/flatcar-master`.

File: build-2345.xml [hunk 2]
```diff
@@ -6,33 +6,33 @@
...
+  <project groups="minilayout" name="flatcar-linux/init" path="src/third_party/init" revision="refs/heads/master" upstream="refs/heads/master"/>
+  <project groups="minilayout" name="flatcar-linux/locksmith" path="src/third_party/locksmith" revision="refs/heads/master" upstream="refs/heads/master"/>
+  <project groups="minilayout" name="flatcar-linux/mantle" path="src/third_party/mantle" revision="refs/heads/master" upstream="refs/heads/master"/>
+  <project groups="minilayout" name="flatcar-linux/mayday" path="src/third_party/mayday" revision="refs/heads/master" upstream="refs/heads/master"/>
+  <project groups="minilayout" name="flatcar-linux/nss-altfiles" path="src/third_party/nss-altfiles" revision="refs/heads/master" upstream="refs/heads/master"/>
+  <project groups="minilayout" name="flatcar-linux/portage-stable" path="src/third_party/portage-stable" revision="refs/heads/flatcar-build-2345.3.1" upstream="refs/heads/flatcar-build-2345.3.1"/>
+  <project groups="minilayout" name="flatcar-linux/scripts" path="src/scripts" revision="refs/heads/flatcar-build-2345.3.1" upstream="refs/heads/flatcar-build-2345.3.1"/>
```
Comment (dongsupark) [inline build-2345.xml]: ditto

File: release.xml [hunk 3]
```diff
@@ -6,33 +6,33 @@
...
+  <project groups="minilayout" name="appc/acbuild" path="src/third_party/appc-acbuild" revision="refs/heads/master" upstream="refs/heads/master"/>
+  <project groups="minilayout" name="appc/spec" path="src/third_party/appc-spec" revision="refs/heads/master" upstream="refs/heads/master"/>
+  <project groups="minilayout" name="flatcar-linux/afterburn" path="src/third_party/afterburn" revision="refs/heads/master" upstream="refs/heads/master"/>
+  <project groups="minilayout" name="flatcar-linux/baselayout" path="src/third_party/baselayout" revision="refs/heads/master" upstream="refs/heads/master"/>
+  <project groups="minilayout" name="flatcar-linux/bootengine" path="src/third_party/bootengine" revision="refs/heads/master" upstream="refs/heads/master"/>
+  <project groups="minilayout" name="flatcar-linux/coreos-cloudinit" path="src/third_party/coreos-cloudinit" revision="refs/heads/master" upstream="refs/heads/master"/>
+  <project groups="minilayout" name="flatcar-linux/coreos-overlay" path="src/third_party/coreos-overlay" revision="refs/heads/flatcar-build-2345.3.1" upstream="refs/heads/flatcar-build-2345.3.1"/>
```
Comment (dongsupark) [inline release.xml:15]: ditto

File: release.xml [hunk 4]
```diff
@@ -6,33 +6,33 @@
...
+  <project groups="minilayout" name="flatcar-linux/init" path="src/third_party/init" revision="refs/heads/master" upstream="refs/heads/master"/>
+  <project groups="minilayout" name="flatcar-linux/locksmith" path="src/third_party/locksmith" revision="refs/heads/master" upstream="refs/heads/master"/>
+  <project groups="minilayout" name="flatcar-linux/mantle" path="src/third_party/mantle" revision="refs/heads/master" upstream="refs/heads/master"/>
+  <project groups="minilayout" name="flatcar-linux/mayday" path="src/third_party/mayday" revision="refs/heads/master" upstream="refs/heads/master"/>
+  <project groups="minilayout" name="flatcar-linux/nss-altfiles" path="src/third_party/nss-altfiles" revision="refs/heads/master" upstream="refs/heads/master"/>
+  <project groups="minilayout" name="flatcar-linux/portage-stable" path="src/third_party/portage-stable" revision="refs/heads/flatcar-build-2345.3.1" upstream="refs/heads/flatcar-build-2345.3.1"/>
+  <project groups="minilayout" name="flatcar-linux/scripts" path="src/scripts" revision="refs/heads/flatcar-build-2345.3.1" upstream="refs/heads/flatcar-build-2345.3.1"/>
```
Comment (dongsupark) [inline release.xml:27]: ditto

**Review by dongsupark — COMMENTED** [review]
`master.xml` looks good.
I have tested it, and it seems to work.

Comments below:

File: build-2345.xml (hunk 1 above)
Comment (pothos) [reply build-2345.xml]: Build `build-2345.xml ` there is to reproduce that build. Otherwise just `master.xml` should be used (which is the default).

Comment (pothos) [reply build-2345.xml]: This file is just used if you add `--manifest-name build-2345` to `cork create`.

Comment (pothos) [reply build-2345.xml]: So, if people want to have the SDK for your latest stable release, they add `--manifest-name release`. In this case they should not land on `flatcar-master` which is the development branch for the next release.

Comment (pothos) [reply build-2345.xml]: Maybe let's even remove `build-2345.xml` as it's the same as `release.xml`.

**Review by dongsupark — APPROVED** [review]
All right.
I am not sure I completely agree.
Though let's merge it for now to unbreak creation of SDK.

**Pull request merged**